WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/WooShipping Create Shipping Labels/WooShipping Split Shipments/WooShippingSplitShipmentsViewModel.swift

```diff
@@ -92,13 +91,62 @@ final class WooShippingSplitShipmentsViewModel: ObservableObject {
         instructions = nil
         dismissedInstructions = true
     }
+
+    func moveSelectedItems(to destination: MoveToShipmentNoticeViewModel.Destination) {
+        moveToNoticeViewModel = nil
+        instructions = nil
+
+        // Step 1: Split items
+        var newShipment = Shipment()
+        var movedItems = [CollapsibleShipmentItemCardViewModel]()
+        for item in currentShipment {
+            let initialQuantity = item.packageItem.quantity.intValue
+            let selectedQuantity = item.numberOfSelectedItems
+            let remainingQuantity = initialQuantity - selectedQuantity
+            if remainingQuantity == 0 {
+                movedItems.append(
+                    CollapsibleShipmentItemCardViewModel(item: item.packageItem, currency: self.order.currency)
+                )
+            } else if selectedQuantity > 0 {
+                let newItem = ShippingLabelPackageItem(copy: item.packageItem, quantity: Decimal(remainingQuantity))
+                newShipment.append(
+                    CollapsibleShipmentItemCardViewModel(item: newItem, currency: self.order.currency)
+                )
+                let movedItem = ShippingLabelPackageItem(copy: item.packageItem, quantity: Decimal(selectedQuantity))
+                movedItems.append(
+                    CollapsibleShipmentItemCardViewModel(item: movedItem, currency: self.order.currency)
+                )
+            } else if selectedQuantity == 0 {
+                newShipment.append(
+                    CollapsibleShipmentItemCardViewModel(item: item.packageItem, currency: self.order.currency)
+                )
+            }
+        }
+
+        // Step 2: Update the current shipment
```

Comment:
Should we remove the current shipment if all items are removed from it? 


https://github.com/user-attachments/assets/1de61982-f455-4280-a651-2e29cf8f82f1

Reply:
You're right! I added the last step to remove empty shipment in de44457d37118029edf052a925de180ee03a3465. Tested in ff4652263620afdbf11c389ce364cad33804d22a.